File: POO/poo1.py
class Coche:
    def __init__(
        self, color: str = "verde", ruedas: int = 4, arrancado: bool = False
    ) -> None:
        self.__color = color
        self.__ruedas = ruedas
        self.__arrancado = arrancado

    def getColor(self) -> str:
        return self.__color

    def setColor(self, color) -> None:
        self.__color = color

    def arrancar_apagar(self) -> None:
        self.__arrancado = not self.__arrancado

    def __str__(self) -> str:
        return f"Color: {self.__color}, Ruedas: {self.__ruedas}, Arrancado: {self.__arrancado}"


coche1 = Coche()

coche1.arrancar_apagar()

print(coche1)

coche1.arrancar_apagar()

print(coche1)

File: POO/EjerciciosPHP/poo1.php
<?php


class Coche
{
  public function __construct(private string $color = 'verde', private int $ruedas = 4, private bool $arrancado = false)
  {
  }

  /**
   * Get the value of color
   */
  public function getColor(): string
  {
    return $this->color;
  }

  /**
   * Get the value of ruedas
   */
  public function getRuedas(): int
  {
    return $this->ruedas;
  }

  /**
   * Get the value of arrancado
   */
  public function getArrancado(): bool
  {
    return $this->arrancado;
  }

  /**
   * Set the value of color
   */
  public function setColor(string $color): self
  {
    $this->color = $color;

    return $this;
  }

  /**
   * Set the value of ruedas
   */
  public function setRuedas(int $ruedas): self
  {
    $this->ruedas = $ruedas;
    return $this;
  }

  /**
   * Set the value of arrancado
   */
  public function setArrancado(bool $arrancado): self
  {
    $this->arrancado = $arrancado;
    return $this;
  }

  public function __toString()
  {
    return "Color: {$this->color}, Ruedas: {$this->ruedas}, Arrancado: " . ($this->arrancado ? 'Sí' : 'No');
  }
}

// Crear una instancia de Coche
$miCoche = new Coche();

// Arrancar el coche
$miCoche->setArrancado(true);

// Imprimir el coche por pantalla
echo $miCoche;

echo '<br>';
echo '<br>';

// Frenar el coche
$miCoche->setArrancado(false);

// Imprimir el coche por pantalla
echo $miCoche;
